### PR TITLE
separate spatial_index function

### PR DESCRIPTION
**Pull request by Casyfill**
separate spatial_index function might be of use for "millions_to_hundreds" sjoint, where we split data into chunks and don't want to generate a rtree.Index for each chunk once again

### CODE REVIEW

**Comment by Casyfill** [comment]
maybe it is better to pass index as **kvargs?

File: geopandas/tools/sjoin.py [hunk 1]
```diff
@@ -24,9 +45,9 @@ def sjoin(left_df, right_df, how='inner', op='intersects',
         Suffix to apply to overlapping column names (left GeoDataFrame).
     rsuffix : string, default 'right'
         Suffix to apply to overlapping column names (right GeoDataFrame).
+    tree_idx : rtree.index for right GeoDataFrame, if None, will be created.
```
Comment (jorisvandenbossche) [inline geopandas/tools/sjoin.py]: Can you put the explanation on a next line (see how it is done for the other kwargs). Typically it should have the form of (following the numpy docstring standard)

```
kwarg : type
    Explanation
```

**Comment by jorisvandenbossche** [comment]
@Casyfill Thanks for the contribution!

I am not that familiar with rtree indexes to judge this, so pinging @mdbartos @perrygeo @jwass who have worked on sjoin before

File: geopandas/tools/sjoin.py (hunk 1 above)
Comment (Casyfill) [inline geopandas/tools/sjoin.py]: Will do!

**Comment by Casyfill** [comment]
Thanks!  I will try to think over a test

**Comment by martinfleis** [comment]
Closed by #789. Thank you @Casyfill for this work even though it was not merged.

**Pull request closed without merging**